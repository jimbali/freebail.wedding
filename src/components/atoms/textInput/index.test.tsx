import { mount, shallow } from 'enzyme'
import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import TextInput from './index'

const mockFn = jest.fn()

describe(TextInput, () => {
  let label = 'Name'
  let name = 'name'
  const onChange = mockFn
  let value = 'Bobbles'

  const instance = () => {
    return <TextInput label={label} name={name} onChange={onChange} value={value}/>
  }

  it('should render without exploding', () => {
    const renderer = createRenderer()
    renderer.render(instance())
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('renders any given label', () => {
    label = 'New Label'
    const textInput = mount(instance())
    expect(textInput.find('label').text()).toEqual(label)
  })

  it('renders any given name', () => {
    name = 'New Name'
    const textInput = mount(instance())
    expect(textInput.find('input').prop('name')).toEqual(name)
  })

  it('renders any given value', () => {
    value = 'New Value'
    const textInput = mount(instance())
    expect(textInput.find('input').prop('defaultValue')).toEqual(value)
  })

  it('calls the mock function when the input is changed', () => {
    const textInput = shallow(instance())
    const input = textInput.find('input')
    const changeEvent = { target: { value: 'Hello' } }

    input.simulate('change', changeEvent)
    expect(mockFn.mock.calls[0][0]).toStrictEqual(changeEvent)
  })
})
