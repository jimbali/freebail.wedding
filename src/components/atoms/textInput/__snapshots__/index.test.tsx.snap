// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render without exploding 1`] = `
<div
  className="row"
>
  <label>
    Name
  </label>
  <input
    className="textInput"
    defaultValue="Bobbles"
    name="name"
    onChange={[MockFunction]}
    type="text"
  />
</div>
`;
